src/object_store/cache.rs

```diff
@@ -365,58 +588,85 @@ impl Display for CachingObjectStore {
 
 #[async_trait]
 impl ObjectStore for CachingObjectStore {
-    async fn put(
-        &self,
-        location: &object_store::path::Path,
-        bytes: Bytes,
-    ) -> object_store::Result<PutResult> {
-        self.inner.put(location, bytes).await
-    }
-
     async fn put_opts(
         &self,
         location: &object_store::path::Path,
         bytes: Bytes,
         opts: PutOptions,
     ) -> object_store::Result<PutResult> {
-        self.inner.put_opts(location, bytes, opts).await
+        let start = Instant::now();
+        let result = self.inner.put_opts(location, bytes, opts).await;
+        self.metrics.log_object_store_outbound_request(
+            start,
+            "put",
+            Some(result.is_ok()),
+        );
+        result
     }
 
     async fn put_multipart(
         &self,
         location: &object_store::path::Path,
     ) -> object_store::Result<(MultipartId, Box<dyn AsyncWrite + Unpin + Send>)> {
-        self.inner.put_multipart(location).await
+        let start = Instant::now();
+        let result = self.inner.put_multipart(location).await;
+        self.metrics.log_object_store_outbound_request(
+            start,
+            "put_multipart",
+            Some(result.is_ok()),
+        );
+        result
     }
 
     async fn abort_multipart(
         &self,
         location: &object_store::path::Path,
         multipart_id: &MultipartId,
     ) -> object_store::Result<()> {
-        self.inner.abort_multipart(location, multipart_id).await
-    }
-
-    async fn get(
```

Comment:
(get is implemented in terms of get_opts nowadays, so just doing get_opts is sufficient)